src/codewhisperer/service/securityScanHandler.ts

```diff
@@ -201,3 +207,46 @@ export async function uploadArtifactToS3(fileName: string, resp: CreateUploadUrl
     })
     getLogger().debug(`StatusCode: ${response.statusCode}`)
 }
+
+export async function patchVulnerableCode({ filePath, issues: [issue] }: AggregatedCodeScanIssue) {
+    const [suggestedFix] = issue.suggestedFixes ?? []
+    if (!suggestedFix) {
+        return
+    }
+    const patch = suggestedFix.code
+    const process = new ChildProcess('patch', [filePath])
```

Comment:
Will it require the file to be saved? If the file is dirty can the patch still be applied?

Reply:
If we apply a patch on unsaved file then if the code where patch is going to be applied is matching with the diff then it will update the vulnerable code similar to what any git pull or apply stash command does. For unsaved content, it will remain unsaved and when user tries to save it, it will show an error notification so that user can either discard the unsaved changes or compare the changes between  saved file after diff applied and the open unsaved file. User can manually override the changes similar to apply git stash command.